src/main/java/com/minecolonies/coremod/colony/buildings/moduleviews/FarmerFieldModuleView.java

```diff
@@ -0,0 +1,144 @@
+package com.minecolonies.coremod.colony.buildings.moduleviews;
+
+import com.ldtteam.blockout.views.Window;
+import com.minecolonies.api.colony.buildings.IBuildingWorkerView;
+import com.minecolonies.api.colony.buildings.modules.AbstractBuildingModuleView;
+import com.minecolonies.coremod.Network;
+import com.minecolonies.coremod.client.gui.modules.FarmerFieldsModuleWindow;
+import com.minecolonies.coremod.network.messages.server.colony.building.farmer.AssignFieldMessage;
+import com.minecolonies.coremod.network.messages.server.colony.building.farmer.AssignmentModeMessage;
+import com.minecolonies.coremod.tileentities.ScarecrowTileEntity;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Client side version of the abstract class for all buildings which require a filterable list of allowed items.
+ */
+public class FarmerFieldModuleView extends AbstractBuildingModuleView
+{
+    /**
+     * Checks if fields should be assigned manually.
+     */
+    private boolean shouldAssignFieldManually;
+
+    /**
+     * Contains a view object of all the fields in the colony.
+     */
+    @NotNull
+    private List<BlockPos> fields = new ArrayList<>();
+
+    /**
+     * The amount of fields the farmer owns.
+     */
+    private int amountOfFields;
+
+    @Override
+    public String getDesc()
+    {
+        return "com.minecolonies.coremod.gui.workerhuts.farmerhut.fields";
+    }
+
+    @Override
+    public void deserialize(@NotNull final PacketBuffer buf)
+    {
+        fields = new ArrayList<>();
+
+        shouldAssignFieldManually = buf.readBoolean();
+        final int size = buf.readInt();
+        for (int i = 1; i <= size; i++)
+        {
+            @NotNull final BlockPos pos = buf.readBlockPos();
+            fields.add(pos);
+        }
+        amountOfFields = buf.readInt();
+    }
+
+    /**
+     * Should the farmer be assigned manually to the fields.
+     *
+     * @return true if yes.
+     */
+    public boolean assignFieldManually()
+    {
+        return shouldAssignFieldManually;
+    }
+
+    /**
+     * Getter of the fields list.
+     *
+     * @return an unmodifiable List.
+     */
+    @NotNull
+    public List<BlockPos> getFields()
+    {
+        return Collections.unmodifiableList(fields);
```

Comment:
could just make the internal list immutable instead of wrapping a list that never gets modified